app/src/org/commcare/views/widgets/ImageWidget.java

```diff
@@ -157,14 +157,16 @@ public void onClick(View v) {
             int screenWidth = display.getWidth();
             int screenHeight = display.getHeight();
 
-            // Check if we have a raw folder, and if so pull the image to display from there
+            File imageBeingSubmitted = new File(mInstanceFolder + "/" + mBinaryName);
+
+            // If there is an image in the raw folder, use that as the display image, since it is
+            // better quality
             File toDisplay = new File(mInstanceFolder + "/raw/" + mBinaryName);
             if (!toDisplay.exists()) {
-                // Otherwise, just use the image in the instance folder
-                toDisplay = new File(mInstanceFolder + "/" + mBinaryName);
+                toDisplay = imageBeingSubmitted;
             }
 
-            checkFileSize(toDisplay);
+            checkFileSize(imageBeingSubmitted);
```

Comment:
This is the fix -- A while back, I made it so that even when image capture resizing is being used, the original full-sized image is still the one that is displayed in the app, just because there isn't any reason not to and it will look nicer. This means that `checkFileSize()` needs to be called on the image in the instance folder that is going to ultimately be submitted, not the display image, since they are potentially different.